ddl/db_test.go

```diff
@@ -7771,3 +7772,34 @@ func (s *testSerialDBSuite) TestAddGeneratedColumnAndInsert(c *C) {
 	tk.MustQuery("select * from t1 order by a").Check(testkit.Rows("4 5", "10 11"))
 	c.Assert(checkErr, IsNil)
 }
+
+func (s *testDBSuite1) TestGetTimeZone(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+
+	testCases := []struct {
+		tzSQL  string
+		tzStr  string
+		tzName string
+		offset int
+		err    error
+	}{
+		{"set time_zone = '+00:00'", "", "UTC", 0, nil},
+		{"set time_zone = '-00:00'", "", "UTC", 0, nil},
+		{"set time_zone = 'UTC'", "UTC", "UTC", 0, nil},
+		{"set time_zone = '+05:00'", "", "UTC", 18000, nil},
+		{"set time_zone = '-08:00'", "", "UTC", -28800, nil},
+		{"set time_zone = 'Asia/Shanghai'", "Asia/Shanghai", "Asia/Shanghai", 0, nil},
+		{"set time_zone = 'SYSTEM'", "Asia/Shanghai", "Asia/Shanghai", 0, nil},
+		{"set time_zone = DEFAULT", "Asia/Shanghai", "Asia/Shanghai", 0, nil},
+		{"set time_zone = 'GMT'", "GMT", "GMT", 0, nil},
+		{"set time_zone = 'EST'", "EST", "EST", 0, nil},
```

Comment:
It also has `offset=39600`.  I think this is also OK, but it's not as simple as the way you suggest.

Reply:
I think that is one of very few time zone location that the test would fail for, but it is a very rare time zone, with just a half-hour DST, but it makes a great test 😃  so it is not a fixed offset.

Thank you for taking in my suggestion!